feat(lib): add `keep_prefix` functionality for long keys

HanadaLee opened the pull request:
Introduce `keep_prefix` parameter for `get` and `set` methods. 

The `keep_prefix` preserves the first 479 bytes of keys longer than 479 bytes, appending the SHA-256 hash to maintain prefix search capability with `lmdb.prefix()` while complying with LMDB's 511-byte key size limit.

This transformation occurs specifically when keys exceed 479 bytes.

If off, when the key exceeds LMDB's 511-byte key size limit, the entire key is automatically hashed with sha-256 and used as the actual key.

Implement #67

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Kong/lua-resty-lmdb?pullRequest=68) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Kong/lua-resty-lmdb?pullRequest=68) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Kong/lua-resty-lmdb?pullRequest=68) it.</sub>

chronolaw commented:
Could we add some test cases for this feature?

Oyami-Srk reviewed (commented):
I like the idea! Could we have some test cases on this? Thanks for the contribution!

HanadaLee commented:
> I like the idea! Could we have some test cases on this? Thanks for the contribution!

Ok, I'll try to add some test cases

HanadaLee committed:
feat(lib): add `keep_prefix` functionality for long keys (#67)

Introduce `keep_prefix` parameter for `get` and `set` methods. The `keep_prefix` preserves the first 479 bytes of keys longer than 479 bytes, appending the SHA-256 hash to maintain prefix search capability with `lmdb.prefix()` while complying with LMDB's 511-byte key size limit. This transformation occurs specifically when keys exceed 479 bytes. If off, when the key exceeds LMDB's 511-byte key size limit, the entire key is automatically hashed with sha-256 and used as the actual key.

chronolaw commented on t/08-max-key-size.t:
```diff
@@ -149,3 +149,214 @@ value2
+[crit]
+
+
+=== TEST 9: key size is 416 set() / get() with keep_prefix
```
Why did we choice `416`? any reason?

HanadaLee replied:
Previously there was a test case with a length of 448, but I don't know what it does. For consistency, I also provided this test case with a length of 448 - 32 = 416.
